src/condor_starter.V6.1/docker_proc.cpp

```diff
@@ -246,6 +328,34 @@ int DockerProc::StartJob() {
 	return TRUE;
 }
 
+int
+DockerProc::PullImage() {
+	CondorError err;
+	int pullReaperId = daemonCore->Register_Reaper("PullReaper", (ReaperHandlercpp)&DockerProc::PullReaper,
+			"PullReaper", this);
+	int r = DockerAPI::pullImage(imageName, credentials_dir(), starter->GetWorkingDir(0), *JobAd, pullReaperId, err);
+	if (r == 0) {
+		return TRUE;
+	} else {
+		return FALSE;
+	}
+}
+
+int
+DockerProc::PullReaper(int pid, int status) {
+	dprintf(D_FULLDEBUG, "DockerProc::pullReaper fired for pid %d with status %d\n", pid, status);
+	if (status == 0) {
+		return LaunchContainer();
```

Comment:
What happens if LaunchContainer() returns FALSE? I don't think the starter is going to react.